cmd/apiserver/server.go

```diff
@@ -60,12 +57,89 @@ func (s *server) createCalculationBulk(c *gin.Context) {
 	}
 
 	if err := s.client.Create(s.ctx, bulk); err != nil {
-		responseError(c, "couldn't create calculation", err)
+		responseError(c, "couldn't create calculation bulk", err)
 	} else {
 		c.JSON(http.StatusOK, gin.H{"data": bulk})
 	}
 }
 
+func (s *server) createWorkerPool(c *gin.Context) {
+	body, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		responseError(c, "couldn't read body", err)
+		return
+	}
+
+	workerPool := &workersv1.WorkerPool{}
+	if err := json.Unmarshal(body, &workerPool); err != nil {
+		responseError(c, "couldn't unmarshal body", err)
+	}
+
+	workerPool.Name = fmt.Sprintf("workerpool-%s", workerPool.Name)
+
+	workerPoolList := &workersv1.WorkerPoolList{}
+	if err := s.client.List(s.ctx, workerPoolList); err != nil {
+		responseError(c, "couldn't get the list of workerpools", err)
+	} else {
+		for _, wp := range workerPoolList.Items {
+			if wp.Name == workerPool.Name {
+				responseError(c, "workerpool with entered name already exists", err)
+				return
+			}
+		}
+	}
+
+	s.logger.Info("Creating the workerpool...")
+
+	if err := s.client.Create(s.ctx, workerPool); err != nil {
+		responseError(c, "couldn't create calculation bulk", err)
+	} else {
+		c.JSON(http.StatusOK, gin.H{"data": workerPool})
+	}
+}
+
+func (s *server) deleteCalculationBulk(c *gin.Context) {
+	calcBulkName := c.Param("id")
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		bulk := &bulkv1.CalculationBulk{}
+		err := s.client.Get(s.ctx, ctrlruntimeclient.ObjectKey{Namespace: s.namespace, Name: calcBulkName}, bulk)
+		if err != nil {
+			responseError(c, fmt.Sprintf("failed to get the calculation bulk %s", calcBulkName), err)
+		}
+
+		if err := s.client.Delete(s.ctx, bulk); err != nil {
+			responseError(c, fmt.Sprintf("failed to delete the calculation bulk %s", calcBulkName), err)
+			return err
+		} else {
+			c.JSON(http.StatusOK, response(fmt.Sprintf("calculation bulk %s has been deleted", calcBulkName), http.StatusOK))
+		}
+		return nil
+	}); err != nil {
+		responseError(c, "retryOnConflict method failed", err)
```

Comment:
We don't want the user to know that. We should send a 500 Internal server error here.